File: hw1/Codes/Oztunc_Zeynep.php
<?php 
//php 7.2.24
//initialize
$newArr = array("a"=>"20", "b"=>"40", "c"=>"60");
print_r($newArr);

//get the value for a given key
echo "The value for the key a is: ", $newArr["a"],"\n" ;

// add a new element
$newArr["d"]="80";
print_r($newArr);

//remove an element
unset($newArr['b']);
print_r($newArr);

//modify the value of an existing array
$newArr["d"]="160";
print_r($newArr);

//search for the existence of a key
print_r("Does the key c exist? \n");
if (array_key_exists("c",$newArr))
{
    echo "Key exists \n";
}
else{
    echo "Key does not exist \n";
}
//search for an existence of a value
if( in_array( "60" ,$newArr ) )
{
    echo "The value exists\n";
}
else{
    echo "The value does not exist \n ";
}

//loop through the array
foreach ($newArr as $key => $value) {
 //echo $key;
    foo($key,$newArr);
}

//prints the key-value pair
function foo($x,$newArr) {
    echo "The value for the key ". $x." is ". $newArr[$x]."\n";
} 
?>

File: hw1/Codes/Oztunc_Zeynep.py
#python 3.6.9

#initialize
newArr = {'a': 10, 'b': 20, 'c':30,'d':40}
print(newArr)

#get the value for a given key
print(newArr["a"])

# add a new element
newArr["e"] = "50"
print(newArr)

#remove an element
del newArr["e"]
print(newArr)

#modify the value of an existing element
newArr["a"] = "11"
print(newArr)

#search for the existence of a key
if "b" in newArr:
 print ("The key exists")
else:
  print ("Key does not exist")

#search for an existence of a value
if 30 in newArr.values():
 print ("The value exists")
else:
  print ("The value does not exist")
    
#prints the key-value pair
def foo(key):
     print(f"The value for the key ", key ," is ", newArr[key])
        
#loop through the array
for val in newArr:
    foo(val)
 
